src/metabase/public_settings.clj

```diff
@@ -809,46 +809,78 @@ See [fonts](../configuring-metabase/fonts.md).")
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Deprecated uploads settings begin
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;; These settings were removed in v50 and will be removed in a future release. They are kept here for backwards compatibility
+;; to avoid breaking existing installations that may have set these values with environment variables, or via the config file.
+
+(defsetting uploads-enabled
+  (deferred-tru "Whether or not uploads are enabled")
+  :visibility :authenticated
+  :export?    false
+  :type       :boolean
+  :default    false
+  :getter     (fn []  (log/warn "'uploads-enabled' has been removed; use 'uploads_enabled' on the database instead"))
+  :setter     (fn [_] (log/warn "'uploads-enabled' has been removed; use 'uploads_enabled' on the database instead")))
 
 (defsetting uploads-database-id
   (deferred-tru "Database ID for uploads")
-  :visibility :authenticated
-  :export?    true
+  :visibility :internal
+  :export?    false
   :type       :integer
-  :audit      :getter
-  :setter     set-uploads-database-id!)
+  :getter     (fn []  (log/warn "'uploads-database-id' has been removed; use 'uploads_enabled' on the database instead"))
+  :setter     (fn [_] (log/warn "'uploads-database-id' has been removed; use 'uploads_enabled' on the database instead")))
 
 (defsetting uploads-schema-name
   (deferred-tru "Schema name for uploads")
-  :visibility :authenticated
-  :export?    true
+  :visibility :internal
+  :export?    false
   :type       :string
-  :audit      :getter)
+  :getter     (fn []  (log/warn "'uploads-schema-name' has been removed; use 'uploads_schema_name' on the database instead"))
+  :setter     (fn [_] (log/warn "'uploads-schema-name' has been removed; use 'uploads_schema_name' on the database instead")))
 
 (defsetting uploads-table-prefix
   (deferred-tru "Prefix for upload table names")
-  :visibility :authenticated
+  :visibility :internal
+  :export?    false
   :type       :string
-  :audit      :getter)
+  :getter     (fn []  (log/warn "'uploads-table-prefix' has been removed; use 'uploads_table_prefix' on the database instead"))
+  :setter     (fn [_] (log/warn "'uploads-table-prefix' has been removed; use 'uploads_table_prefix' on the database instead")))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Deprecated uploads settings end
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defsetting uploads-settings
+  (deferred-tru "Upload settings")
+  :visibility :authenticated
+  :export?    false ; the data is exported with a database export, so we don't need to export a setting
+  :type       :json
+  :audit      :getter
+  :getter     (fn []
+                (let [db (t2/select-one :model/Database :uploads_enabled true)]
+                  {:db_id        (:id db)
+                   :schema_name  (:uploads_schema_name db)
+                   :table_prefix (:uploads_table_prefix db)}))
+  :setter     (fn [{:keys [db_id schema_name table_prefix]}]
+                (cond
+                  (nil? db_id)
```

Comment:
should we check write perm in this case too?

Reply:
Huh, that's a great question. We currently don't check any write perms when disabling uploads, but maybe we should. We probably didn't realise this before because uploads weren't a database-level setting. It's a very niche problem though: settings managers can disable uploads for databases that they don't have write permissions for. I'll make a fix and test it works but I wouldn't block this PR on it